opencl-aot/source/main.cpp

```diff
@@ -253,17 +265,46 @@ getCompilerBuildLog(const CLProgramUPtr &ProgramUPtr, cl_device_id Device) {
 
 int main(int Argc, char *Argv[]) {
   // step 0: set command line options
-  cl::opt<std::string> OptInputBinary(
-      cl::Positional, cl::Required,
-      cl::desc("<input SPIR-V or OpenCL program binary>"),
+  cl::opt<std::string> OptInputFile(
+      cl::Positional,
+      cl::desc("<input SPIR-V or OpenCL program binary or OpenCL C source>"),
       cl::value_desc("filename"));
+  cl::opt<std::string> OptOutputElf(
+      "o", cl::init("aot-output.bin"),
+      cl::desc("Specify the output OpenCL program binary filename"),
+      cl::value_desc("filename"));
+
   cl::opt<DeviceType> OptDevice(
       "device", cl::Required, cl::desc("Set target device type:"),
       cl::values(
           clEnumVal(cpu, "Intel(R) processor device"),
           clEnumVal(gpu, "Intel(R) Processor Graphics device"),
           clEnumVal(fpga_fast_emu,
                     "Intel(R) FPGA Emulation Platform for OpenCL(TM) device")));
+
+  enum Commands { build, compile, link };
+  cl::opt<Commands> OptCommand(
+      "cmd", cl::desc("Command"),
+      cl::values(
+          clEnumVal(
+              build,
+              "Build executable IR from OpenCL source or OpenCL .obj file"),
+          clEnumVal(compile, "Compile from OpenCL source to .obj"),
+          clEnumVal(link, "Link OpenCL .obj file(s)")));
+
+  // These options are only needed for compatibility with old tool.
+  // TODO: Delete these options once transition to a regular -o and regular
+  // input argument is complete.
```

Comment:
aoc tool will be replaced by this tool.  This is why link add these changes.

Reply:
I thought there will be some intermediate step, in which aoc calls opencl-aot instead of aocl-ioc64.
In this case, it is even easier - as separate patch, Clang driver should be updated somewhere [here](https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/SYCL.cpp#L304) in the same way as it was done for opencl-aot integration for x86 [here](https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/SYCL.cpp#L462).

Reply:
I'm not familiar with Clang driver. My suggestion is to implement these deprecated options in intel customizations style if community doesn't accept.

Reply:
We need to understand all the use-cases here: in intel/llvm we indeed could avoid updating `opencl-aot` tool and just update the driver to use it instead of `aocl-ioc64`.

However, if this tool is used in some other workflow/toolchain somewhere (like a downstream project) and we are not ready to change that workflow to pass new options yet, then this patch is indeed needed. But please note that we will do extra work twice: in this PR we introducing compatibility option and later we will have another PR which removes them. If we actually could update our usage model of the tool, it would be better (1 task instead of 3)